feat: allow auto/pre-selected extension

kohanyirobert opened the pull request:
Create a rough version of automatically selecting/appending extensions.
No backend change, only frontend.
I haven't tested it thoroughly unfortunately.
With all the flags and switches going on I'm not sure it works logically all the time.

(I've messed up something, but ultimately wanted to link together the PR and #254)

kohanyirobert commented:
I worked on this in a devcontainer, hence the extra files, let me know if you don't need those.

marcopiovanello commented on frontend/src/atoms/settings.ts:
```diff
@@ -59,12 +60,12 @@ export const themeState = atomWithStorage<Theme>(
 export const serverAddressState = atomWithStorage<string>(
   'server-addr',
-  localStorage.getItem('server-addr') || window.location.hostname
+  'localhost' // || localStorage.getItem('server-addr') || window.location.hostname
```
revert to localstorage + fallback. localhost as default will break first installations.

kohanyirobert replied:
Ah yeah, sorry about this. I didn't know how to provide these for the frontend so I just hardcoded this for testing.

While developing with "pnpm run dev" how to tell where the backend is for the frontend? (Aside from hard-coding this in the code.) Manually adding it into localStorage? What's the "suggested" development loop?

marcopiovanello commented on frontend/src/atoms/settings.ts:
```diff
@@ -59,12 +60,12 @@ export const themeState = atomWithStorage<Theme>(
 export const serverPortState = atomWithStorage<number>(
   'server-port',
-  Number(localStorage.getItem('server-port')) || Number(window.location.port)
+  12345 // Number(localStorage.getItem('server-port')) || Number(window.location.port)
```
revert to localstorage + fallback. port 12345 as default will break first installations.

kohanyirobert replied:
Reverted

marcopiovanello commented on README.md:
````diff
@@ -13,7 +13,7 @@ Created for the only purpose of *fetching* videos from my server/nas and monitor
 
 ```sh
-docker pull marcobaobao/yt-dlp-webui
+docker pull marcobaobao/yt-dlp-webuiRw
````
probably an oversight, needs to be reverted

kohanyirobert replied:
Yep, sometimes my son is tapping away on my keyboard when I don't look :D Will revert this.

kohanyirobert replied:
Reverted

marcopiovanello reviewed (changes requested):
Translations and React code are fine.

Check the atoms serverAddresState and serverPortState atoms.

kohanyirobert commented:
Reverted my uncanny changes, thanks being on the lookout 🙏 

A few questions:
1 .While developing with "pnpm run dev" how to tell where the backend is for the frontend? (Aside from hard-coding this in the code.) Manually adding it into localStorage? What's the "suggested" development loop?
2. Should I add some strings about using devcontainers to the README?
3. I added `--host 0.0.0.0` to the `package.json` as an argument for the `dev` script, reason being developing in a container I wasn't able to access the service outside on my localhost. Is that okay?

marcopiovanello commented:
> Reverted my uncanny changes, thanks being on the lookout 🙏
> 
> A few questions: 1 .While developing with "pnpm run dev" how to tell where the backend is for the frontend? (Aside from hard-coding this in the code.) Manually adding it into localStorage? What's the "suggested" development loop? 2. Should I add some strings about using devcontainers to the README? 3. I added `--host 0.0.0.0` to the `package.json` as an argument for the `dev` script, reason being developing in a container I wasn't able to access the service outside on my localhost. Is that okay?

1. it's not yet implemented... it's by default on port 3033 on the first boot, then editable in the settings section.
2. I think it's fine, I develop mainly on linux and macos which hasn't a firewall by default. Maybe it's a "problem" in windows where the firewall asks you to "open" the port.
